cmd/node-termination-handler.go

```diff
@@ -113,11 +113,12 @@ func main() {
 		nthConfig.Print()
 		log.Fatal().Err(err).Msg("Unable to instantiate probes service,")
 	}
+        imdsDisabled := nthConfig.EnableSQSTerminationDraining
 
 	imds := ec2metadata.New(nthConfig.MetadataURL, nthConfig.MetadataTries)
 
 	interruptionEventStore := interruptioneventstore.New(nthConfig)
-	nodeMetadata := imds.GetNodeMetadata()
+	nodeMetadata := imds.GetNodeMetadata(!imdsDisabled)
```

Comment:
Given that both the variable and the function parameter are called "disabled", shouldn't this line be the following?

```suggestion
	nodeMetadata := imds.GetNodeMetadata(imdsDisabled)
```

Reply:
To clarify: GetNodeMetadata() does get called in queue mode with the current code as per this MR. Unfortunately my Gelang skills are not sufficient to come up with an elegant refactoring that truly separates the code parts.